### ISSUE TITLE
Suppress Lint/AssignmentInCondition in a block

### ISSUE THREAD

**Issue by mruzicka** [rubocop-hq/rubocop#5604]
I have the following logic in my command line options parsing code:
```ruby {.line-numbers}
 1: if @argv.empty? ||
 2:    begin
 3:      remaining = parser.permute(@argv)
 4: 
 5:      # Set the mode
 6:      options[:mode] = remaining.shift
 7: 
 8:      if options[:mode] == 'help'
 9:        options[:help] = true
10:        options[:mode] = remaining.shift
11:      end
12: 
13:      # Update the parser for the new mode
14:      parser.update
15: 
16:      # Value of the block
17:      options[:help]
18:    end
19: then # rubocop:disable Style/MultilineIfThen
20:   puts parser.help
21:   exit 0
22: end
```
Please note the `begin` ... `end` block in the condition.

--------

## Expected behavior

RuboCop realizes there is a block in the condition and doesn't issue any `Lint/AssignmentInCondition` warnings for the code inside the block.

## Actual behavior

RuboCop probably misses the fact that there is a block in the condition and issues the following warnings:
```
cli.rb:3:16: W: Lint/AssignmentInCondition: Use == if you meant to do a comparison or wrap the expression in parentheses to indicate you meant to assign in a condition. (https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)
     remaining = parser.permute(@argv)
               ^
cli.rb:6:21: W: Lint/AssignmentInCondition: Use == if you meant to do a comparison or wrap the expression in parentheses to indicate you meant to assign in a condition. (https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)
     options[:mode] = remaining.shift
                    ^
cli.rb:9:23: W: Lint/AssignmentInCondition: Use == if you meant to do a comparison or wrap the expression in parentheses to indicate you meant to assign in a condition. (https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)
       options[:help] = true
                      ^
cli.rb:10:23: W: Lint/AssignmentInCondition: Use == if you meant to do a comparison or wrap the expression in parentheses to indicate you meant to assign in a condition. (https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)
       options[:mode] = remaining.shift
                      ^
```

## Steps to reproduce the problem

Run RuboCop on the code snippet given above.

## RuboCop version

0.50.0 (using Parser 2.4.0.2, running on ruby 2.3.3 x86_64-darwin17)

**Comment by mikegee** [comment]
> RuboCop realizes there is a block in the condition and doesn't issue any `Lint/AssignmentInCondition` warnings for the code inside the block.

I guess so. This code example makes me think we need a new cop that disallows `begin/end` in a conditional, though.
